### DISCUSSION TITLE
Is it possible to spawn a new window in a specific domain?

### DISCUSSION THREAD

**Discussion by cole-h** [wezterm/wezterm#2730] [Q&A] [answered]
Essentially, I'm hoping to make my <kbd>Ctrl Shift n</kbd> keybind always spawn a window in a specific domain.

Right now, I have a `scratch` domain setup that, with some plumbing in various places, will move any wezterm connected to that domain to my swaywm scratchpad. However, I don't want to hit <kbd>Ctrl Shift n</kbd> and continue to spawn `scratch` domain windows -- I want to use the `default_domain` to prevent this (since my plumbing relies on the current window's domain).

I (naively) tried to use an `action_callback` and "change" the pane's domain_name, but realized the pane given to the callback doesn't have the `domain_name` field accessible (I believe the difference is between the `Pane` struct and the `PaneInformation` struct), and this likely wouldn't work anyways.

**Comment by wez** [comment, marked as answer]
Take a look at [SpawnCommandInNewWindow](https://wezfurlong.org/wezterm/config/lua/keyassignment/SpawnCommandInNewWindow.html).  It accepts a [SpawnCommand](https://wezfurlong.org/wezterm/config/lua/SpawnCommand.html) that allows you to specify a number of things about what gets spawned into the new window.

**Reply by cole-h** [reply]
Perfect, thank you!
